src/argilla_sdk/records/_io/_datasets.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2024-present, Argilla, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC
+from typing import Dict, List, TYPE_CHECKING, Union, Iterable
+
+from datasets import Dataset
+
+from argilla_sdk.records._io._generic import GenericIOMixin
+
+if TYPE_CHECKING:
+    from argilla_sdk import Record
+
+
+class HFDatasetsIOMixin(Iterable["Record"], ABC):
```

Comment:
I think we can define helper IO classes instead of mixins to convert "something" into records or records into "something", with the same principle of splitting them by "format" (the "something"). Let me show an example.

We can redefine datasets IO as:
```python
class HFDatasetsIO:
    @staticmethod
    def to_datasets(records: List[Record]) -> Dataset:
      ... 
 
    @staticmethod
    def from_datasets(dataset: Dataset) -> List[Record]:
      ...
```

But in the same way, Json IO as:
```python
class JsonIO:
    @staticmethod
    def to_json(records: List[Record]) -> Path:
      ... 
 
    @staticmethod
    def from_json(path: str) -> List[Record]:
      ...
```

And Generic (or List and Dict) IO:
```python
class GenericIO:
    @staticmethod
    def to_dict(records: List[Record], flatten: bool = False, orient: str = "names") -> Dict[str,List[Any]]:
      ... 
 
    @staticmethod
    def to_list(records:List[Record], flatten: bool = False) -> List[Record]:
      ...
```

And use them in `DatasetRecords` as:

```python

   def _ingest_records(...):
      if isinstance(records, HFDatasets):
         records = HFDatasetsIO.from_datasets()
 

   def to_json():
      records = self(with_....)
      return JsonIO.to_json(records)
```

Reply:
I've re-implemented it like this. I agree that it's an improvement. Thanks.

N.B. `HFDatasetIO` still has `_record_dicts_from_datasets` because that allows it to be combined with mapping in the `add` and `update` . I think that we should maintain this because being able to map columns is more useful than method symmetry.

Reply:
Okay, I see. We can leave it as is. We can revisit once we review the Record.from_mapping and all this stuff. 

Things that could be nice to keep in mind:
1. `from_json` could accept custom JSON format, so a mapping would be great.
2. If we have a `records.from_json` method, maybe it makes sense the `records.from_datasets` method?